Firestore/core/src/firebase/firestore/remote/grpc_connection.mm

```diff
@@ -62,6 +64,13 @@
   RegisterConnectivityMonitor();
 }
 
+void GrpcConnection::Shutdown() {
+  // Fast finish any pending calls. This will not trigger the observers.
+  for (GrpcCallInterface *call : active_calls_) {
+    call->Finish();
```

Comment:
The difference between "finish" and "finish with error" isn't super obvious to me. Consider a hypothetical caller that needs to terminate a call, but not in response to some gRPC error. How do they choose which finish to call? Specifically, when should the caller synthesize an error rather than just calling `Finish`? These names don't suggest an answer.

A few possibly ways out of this are:
  * Rename `Finish` to `Shutdown`: this relies on the system-wide notion that shutdown should avoid callbacks
  * Rename `FinishWithError` to `FinishAndNotify`
  * Rename `Finish` to something like `FinishImmediate` or `FinishSilently`?